src/main.rs

```diff
@@ -1,124 +1,96 @@
-use std::env;
-use std::path::{Path, PathBuf};
 use std::time::Instant;
 
+use cli::{Cli, Command};
 use utils::net::{get_available_port, port_is_available};
 
+use clap::Parser;
+
 mod cli;
 mod cmd;
 mod console;
 mod prompt;
 
 fn main() {
-    let matches = cli::build_cli().get_matches();
-
-    let root_dir = match matches.value_of("root").unwrap() {
-        "." => env::current_dir().unwrap(),
-        path => PathBuf::from(path)
-            .canonicalize()
-            .unwrap_or_else(|_| panic!("Cannot find root directory: {}", path)),
-    };
-    let config_file = match matches.value_of("config") {
-        Some(path) => PathBuf::from(path)
-            .canonicalize()
-            .unwrap_or_else(|_| panic!("Cannot find config file: {}", path)),
-        None => root_dir.join("config.toml"),
-    };
+    let cli = Cli::parse();
+    let root_dir = cli
+        .root
+        .canonicalize()
+        .unwrap_or_else(|_| panic!("Cannot find root directory: {}", cli.root.display()));
```

Comment:
that's not handling the `.` case?

Reply:
Here `cli.root` already has the `.` value if it wasn't specified on the command line, so we handle only the canonicalize operation result, which I think should work for `.`

Reply:
I'm not entirely sure it does work on all platforms, I'd like to be certain before changing it

Reply:
I built a small program to verify that canonicalizing `.` works on windows, but if we wanted to be extra careful, there's the [normpath](https://github.com/dylni/normpath) crate that seems to be saner (also personally confirmed that it works).

Reply:
Let's keep like you have for now, we'll see if it causes an issue